activitychart-basic.html

```diff
@@ -106,29 +106,57 @@
       },
 
       _showLabel: function (d, i) {
+        //using a group element 'label' with an appended text element and a rect element as background to display a label
+
         var polyElement = Polymer.dom(this).parentNode.parentNode.parentNode;
-        d3.select('svg').append('rect')
-					.attr('x', polyElement._spaceSquareHor(d, i) - 3)
-					.attr('y', polyElement._spaceSquareVert(d, i) - 14)
-					.attr('id', 'label-bg-' + i)
-					.attr('fill', 'white')
-					.attr('width', 60)
-					.attr('height', 15);
-
-				d3.select('svg').append('text')
+
+        var label = d3.select('svg').append('g')
+          .attr('id', 'label-' + i);
+
+        //append rect element first so that following text element is displayed above rect
+        label.append('rect');
+
+        var labelText = label.append('text')
 					.text(d + ' at ' + i)
-					.attr('x', polyElement._spaceSquareHor(d, i))
-					.attr('y', polyElement._spaceSquareVert(d, i) - 3)
-					.attr('id', 'label-' + i)
 					.style('font-size', '0.8em')
 					.attr('fill', 'black');
+        //this split is necessary because the position is set relative to the size of the text element
+        labelText
+          .attr('x', polyElement._spaceLabelHor(d, i, labelText.node().getBBox().width))
+          .attr('y', polyElement._spaceLabelVert(d, i));
+
+        //finish rect configuration that was appended previously
+        label.select('rect')
+          .attr('x', labelText.attr('x') - 3)
+  				.attr('y', labelText.attr('y') - 11)
```

Comment:
where do all these hardcoded values (`y -11` etc., rather e.g. `y - something.width`) come from and are they needed like this (probably not)?
Can you maybe rather put them into a var as well and calc them respectively?